cmd/dp-search-query/main.go

```diff
@@ -34,37 +24,30 @@ func main() {
 	// sensitive fields are omitted from config.String().
 	log.Info("config on startup", log.Data{"config": cfg})
 
-	log.Debug("Starting server", log.Data{"Port": cfg.BindAddr, "ElasticSearchUrl": cfg.ElasticSearchAPIURL})
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 
-	// Setup libraries and handlers
-	elasticsearch.Setup(cfg.ElasticSearchAPIURL)
-	errSearch := handlers.SetupSearch()
-	if errSearch != nil {
-		log.ErrorC("Failed to setup search templates", errSearch, log.Data{})
-	}
-	errData := handlers.SetupData()
-	if errData != nil {
-		log.ErrorC("Failed to setup data templates", errData, log.Data{})
-	}
-	errTimeseries := handlers.SetupTimeseries()
-	if errTimeseries != nil {
-		log.ErrorC("Failed to setup timeseries templates", errTimeseries, log.Data{})
-	}
+	apiErrors := make(chan error, 1)
 
-	// Setup web handlers for the search query services
-	router := pat.New()
-	router.Get("/search", handlers.SearchHandler)
-	router.Get("/timeseries/{cdid}", handlers.TimeseriesLookupHandler)
-	router.Get("/data", handlers.DataLookupHandler)
-	router.Get("/healthcheck", handlers.HealthCheckHandlerCreator()) // TODO Replace with healthcheck middleware
-	server = &http.Server{
-		Addr:         cfg.BindAddr,
-		Handler:      router,
-		ReadTimeout:  5 * time.Second,
-		WriteTimeout: 10 * time.Second,
-	}
+	api.CreateAndInitialise(cfg.BindAddr, cfg.ElasticSearchAPIURL, apiErrors)
 
-	if err := server.ListenAndServe(); err != nil {
-		log.ErrorC("Failed to bind to port address", err, log.Data{"Port": cfg.BindAddr})
+	// blocks until a fatal error occurs
+	select {
+	case err := <-apiErrors:
+		log.ErrorC("api error received", err, nil)
+	case <-signals:
+		log.Debug("os signal received", nil)
 	}
+
+	// Gracefully shutdown the application closing any open resources.
+	log.Info(fmt.Sprintf("shutdown with timeout: %s", cfg.GracefulShutdownTimeout), nil)
```

Comment:
As a general rule of thumb the log message should not be dynamic - the variable(s) should added to the `log.Data`. If the messages are static we can search for them more easily when investigating issues.